src/ScottPlot5/ScottPlot5 Cookbook/Recipes/PlotTypes/SignalXY.cs

```diff
@@ -59,7 +59,32 @@ public override void Execute()
             myPlot.Add.SignalXY(xs, ys);
         }
     }
+    
+    public class SignalXYGenericList : RecipeBase
+    {
+        public override string Name => "SignalXY Generic List";
+        public override string Description => "SignalXY plots are a high performance plot type " +
+                                              "for X/Y data in List<Tx,Ty> where the X values are always ascending.";
 
+        [Test]
+        public override void Execute()
+        {
+            // generate sample data with gaps
+            List<double> xList = new();
+            List<double> yList = new();
+            xList.AddRange(Generate.Consecutive());
+            yList.AddRange(Generate.Sin(51));
+
+            // add a SignalXY plot
+            var plt = myPlot.Add.SignalXY(xList, yList);
+            
+            // Extend Data and refresh
+            //xList.AddRange(Generate.Consecutive());
```

Comment:
The required logic to extend the data is like this:

```cs
xs.Add(x);
ys.Add(y);
plt.Data.MaximumIndex++;
WpfPlot1.Refresh();
WpfPlot1.Plot.Axes.AutoScale();
```

I think everything except the increment to `MaximumIndex` field is expected by the user. It might be worth having a `Update()` function, since the `MaximumIndex` field is probably not well known. The other option would be to have `MaximumIndex` become nullable, and it's ignored unless it's specifically set by the user.

We could implement this something like this:

```cs
private int? _maximumIndex = null;
public int MaximumIndex {
  get => _maximumIndex.GetValueOrDefault(Count - 1);
  set {
    _maximumIndex = value;
  }
}
```

Reply:
Oof, it seems like there could be a better design where maximum index is optional, nullable, initialized to infinity, or something so the user doesn't have to do this. I'll try to put something in now. Thanks for pointing this out!